eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -288,9 +330,12 @@ jobs:
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
           goals: ${{ parameters.TestGoals }}
+        # we want to run this when TestFromSource isn't true (which covers normal running when it isn't set) 
+        # OR when ShouldRunSourceTests is true
+        condition: or(ne(variables['TestFromSource'],'true'), eq(variables['ShouldRunSourceTests'],'true'))
```

Comment:
You are going to want to and this condition with "succeeded" https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#succeeded so that we don't end up trying to run this step when some other step already failed.

Reply:
Good catch. Updated with latest commit.